types/block.go

```diff
@@ -100,32 +128,38 @@ func (dah *DataAvailabilityHeader) Hash() []byte {
 	}
 	// The single data root is computed using a simple binary merkle tree.
 	// Effectively being root(rowRoots || columnRoots):
-	return merkle.HashFromByteSlices(slices)
+	dah.hash = merkle.HashFromByteSlices(slices)
+	return dah.hash
 }
 
-func (dah *DataAvailabilityHeader) ToProto() (*tmproto.DataAvailabilityHeader, error) {
+func (dah *DataAvailabilityHeader) ToProto() *tmproto.DataAvailabilityHeader {
 	if dah == nil {
-		return nil, errors.New("nil DataAvailabilityHeader")
+		return nil
```

Comment:
What triggered this change? If a `nil` `DataAvailabilityHeader` is invalid, we want to make sure no-one accidentally creates one and sends it over the wire.

Reply:
I had a use case for nil DAHeader in Vote for nil, as I used DAHeader for block identification in the bulky PR.

Now, I see use cases for nil DAHeader, but indeed not for being sent over the wire, so we need to restrict that. Furthermore, what do you think about panicking in this case? In such a case that's 100% programmer error that will be caught by testing, same as returning the error, but that makes code dirtier imho.

Reply:
IMO, usually, we should prefer erroring instead of panicking but I think you are right that in this case there is no way this can happen and it would be a bug if it did. So yeah, panicking is OK here.

Reply:
@marbar3778 would that be too inconsistent with the other existing `ToProto` methods though?

Reply:
From what I saw, some `ToProto` returns an error, some do not.

Reply:
But anyway, I decided to return an error for you to look at and compare both methods and how do they affect the codebase. I would say that the one with panic does not require so many changes throughout the codebase while still correct. Although I am ok to go both ways.